samples/cellular/lwm2m_client/sample_description.rst

```diff
@@ -397,9 +407,140 @@ CONFIG_SENSOR_MODULE_LIGHT - Configuration for light reading
 CONFIG_SENSOR_MODULE_COLOR - Configuration for color
    This configuration option enables the reading of color values.
 
+.. _lwm2m_configuration_files:
+
+Configuration files
+===================
+
+The sample provides predefined configuration files for typical use cases.
+
+* :file:`prj.conf` - Default configuration file.
+
+LwM2M Device management server:
+
+* :file:`overlay-leshan-bootstrap.conf` - Enables LwM2M bootstrap support with Leshan demo server.
+* :file:`overlay-avsystem.conf` - Uses `Coiote Device Management`_ server.
+* :file:`overlay-avsystem-bootstrap.conf` - Uses Coiote in bootstrap mode.
+
+NB-IoT:
+
+* :file:`overlay-nbiot.conf` - Enables the use of NB-IoT.
+
+LwM2M v1.1:
+
+* :file:`overlay-lwm2m-1.1.conf` - Enables LwM2M v1.1 protocol version.
+
+Firmware update:
+
+* :file:`overlay-adv-firmware.conf` - Enables experimental Advanced Firmware Update object.
+* :file:`overlay-fota_helper.conf` - Enables faster response for evaluating FOTA.
+
+Location assintance:
+
+* :file:`overlay-assist-agnss.conf` - Enables A-GNSS assistance.
+* :file:`overlay-assist-cell.conf` - Enables cell-based location assistance.
+* :file:`overlay-assist-pgps.conf` - Enables P-GPS assistance in the sample.
+
+Location service requirerequire `Coiote Device Management`_ server and LwM2M v1.1.
+
+DTLS optimications:
+
+* :file:`overlay-dtls-cid.conf` - Enables DTLS Connection Identifier.
+
+Power savings:
+
+* :file:`overlay-aggressive-psm.con` - Enables optimized PSM setup and proprietary PSM mode.
+* :file:`overlay-lowpower.conf` - Disables serial console to bring the power consumption down.
+
+LwM2M v1.1 conformance testing:
+
+* :file:`overlay-lwm2m-1.1-core-interop.conf` - Allows running of Core Specific Objects Test cases.
+* :file:`overlay-lwm2m-1.1-object-interop.conf` - Allows running of Additional Objects Test cases.
+
+
 Additional configuration
 ========================
 
+Default configuration
```

Comment:
Add an intro sentence before the sub section.. Following are some additional configurations that could be used while using the sample.

Reply:
No intro sentence yet.